examples/graphql_examples/tests/src/Kernel/Schema/ExampleSchemaTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Drupal\Tests\graphql_example\Kernel;
+
+use Drupal\node\Entity\Node;
+use Drupal\node\Entity\NodeType;
+use Drupal\Tests\graphql\Kernel\GraphQLTestBase;
+use Drupal\user\Entity\User;
+
+/**
+ * Runs unit tests agains the `example` schema defined in `graphql_examples`.
+ *
+ * @group graphql
+ */
+class ExampleSchemaTest extends GraphQLTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = ['graphql_examples'];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() :void {
+    parent::setUp();
+    // Create the "article" node type since the schema relies on it.
+    NodeType::create([
+      'type' => 'article',
+      'name' => 'Article',
+    ]);
+
+    // Create a test-server that uses the schema plugin defined in this module.
+    $this->createTestServer('example', '/graphql');
+  }
+
+  /**
+   * Test the example schema for article listing.
+   */
+  public function testExampleSchema() : void {
+    // Create two authors.
+    $userA = User::create([
+      'name' => 'A',
+    ]);
+    $userA->save();
+
+    $userB = User::create([
+      'name' => 'B',
+    ]);
+    $userB->save();
+
+    // Create three articles.
+    Node::create([
+      'type' => 'article',
+      'title' => 'One',
+      'uid' => $userA->id(),
+    ])->save();
+
+    Node::create([
+      'type' => 'article',
+      'title' => 'Two',
+      'uid' => $userB->id(),
+    ])->save();
+
+    Node::create([
+      'type' => 'article',
+      'title' => 'Three',
+      'uid' => $userA->id(),
+    ])->save();
+
+    // Execute the query and run assertions against its response content.
+    $response = $this->query('{ articles { total, items { title, author } } }');
+    $content = json_decode($response->getContent(), TRUE);
+    $this->assertEquals([
+      'data' => [
+        'articles' => [
+          'total' => 3,
+          'items' => [
+            ['title' => 'ONE', 'author' => 'A'],
+            ['title' => 'TWO', 'author' => 'B'],
+            ['title' => 'THREE', 'author' => 'A'],
+          ],
+        ],
+      ],
+    ], $content);
```

Comment:
Instead of `$this->query` I think we should use `$this->assertResults`.

The benefits are that it bypasses the HTTP stack which is slightly faster. It also handles the response decoding for you and allows you to test for caching metadata.

```suggestion
    $this->assertResults(
      '{ articles { total, items { title, author } } }',
      [],
      [
        'data' => [
          'articles' => [
            'total' => 3,
            'items' => [
              ['title' => 'ONE', 'author' => 'A'],
              ['title' => 'TWO', 'author' => 'B'],
              ['title' => 'THREE', 'author' => 'A'],
            ],
          ],
        ],
      ]
    );
```

Reply:
You are right @Kingdutch ! 🤦 Thanks for the input.